src/Model/ModelManager.php

```diff
@@ -51,7 +51,7 @@ class ModelManager implements ModelManagerInterface, LockInterface
      */
     protected $cache = [];
 
-    public function __construct(RegistryInterface $registry)
+    public function __construct(ManagerRegistry $registry)
```

Comment:
this was the only issue I noticed:

https://github.com/doctrine/DoctrineBundle/blob/master/UPGRADE-2.0.md#registry

Reply:
This will work for both major versions?

Reply:
Yes. I tested it with 1.10 and 1.11 as well.

Reply:
This is a BC-break. The type declaration should be dropped, and a deprecation triggered to tell people they have to pass an instance of `ManagerRegistry` if they didn't. Also, a `NEXT_MAJOR` note should be added to remind us to add a type declaration on the master branch.

Reply:
Are you sure this is a BC break?

I think all instances of `RegistryInterface` are actually instance of `ManagerRegistry`?

Reply:
The change is this one:

https://github.com/doctrine/DoctrineBundle/commit/43a1929fb193ce166affabfb1d7ec2763594b2ac

Which extends this one:

https://github.com/symfony/symfony/blob/master/src/Symfony/Bridge/Doctrine/ManagerRegistry.php#L23

Reply:
@dmaicher the user could write a decorator for the manager registry, so technically it's a BC break, yeah.

Reply:
Ok indeed :+1: this case would break.

Reply:
But wait. Would it really break?

If I have something like this:

```php
class MyManager extends ModelManager
{
    public function __construct(RegistryInterface $registry)
    {
        parent::__construct($registry);
    }
}
```

It will still work fine since we made it more loose on the parent class.

`RegistryInterface` actually extends `ManagerRegistry`.

Or am I missing something?

Reply:
You can do decorator for `RegistryInterface`, or you can implement your `RegistryInterface`.

Reply:
Which still means I have an instance of `ManagerRegistry` then?

https://github.com/symfony/symfony/blob/4.4/src/Symfony/Bridge/Doctrine/RegistryInterface.php#L24

Reply:
You mean to change for this -https://github.com/doctrine/persistence/blob/master/lib/Doctrine/Persistence/ManagerRegistry.php ?

Reply:
I fail to see how the change is a BC break. Maybe you have an example?

Reply:
https://symfony.com/doc/current/service_container/service_decoration.html
You can decorate your interface.
But interesting why is not any tests is failed?

Reply:
Your change maybe don't throw exception, because `RegistryInterface` has been extend from `ManagerRegistry`, but if someone decorate `RegistryInterface` they can be have unexpected behavior.

Reply:
Service decoration is exactly what I had in mind. The decorator would not extend the concrete class. In fact, it wouldn't even know about it, but only about the interface.

Reply:
Remove type declaration or change to parent class, will be have 'little' break BC. I don't have idea how it can do without break BC.

Reply:
Maybe to add deprecation triggered on `RegistryInterface`? I think is can help to detected `unexpected behavior `? @greg0ire What you think about it?

Reply:
Removing the type declaration is absolutely not a BC break since this is a constructor. I think I have been clear on what needs to be done in my first message

Reply:
@greg0ire I still fail to see how this can be a BC break.

The now deprecated `RegistryInterface` extends Doctrine's `ManagerRegistry` interface:

https://github.com/symfony/symfony/blob/4.4/src/Symfony/Bridge/Doctrine/RegistryInterface.php#L24

So this will work even for any custom implementation of `RegistryInterface` as the tests show.

Reply:
Oh right, I didn't have that piece of the puzzle (I'm at the SfCon and that kind of inheritance is hard to check with just a phone). Everything should be fine indeed :)